src/org/mozilla/javascript/NativeJavaMap.java

```diff
@@ -25,63 +25,86 @@ public String getClassName() {
         return "JavaMap";
     }
 
-
     @Override
     public boolean has(String name, Scriptable start) {
-        if (map.containsKey(name)) {
-            return true;
+        Context cx = Context.getContext();
```

Comment:
"getContext" will throw a RuntimeError if there is no Context on the stack. That's usually incorrect but I know that there's code out there that might do it. Could you please replace these tests with:

Context cx = Context.getCurrentContext();
if (cx != null && cx.hasFeature(Context.FEATURE_ENABLE_JAVA_MAP_ACCESS))

(or make a helper function in Context or ScriptRuntime since we do that ll the time!)

Reply:
dc43827
I fixed `NativeJavaMap` and `NativeJavaList` since I based it on #713.